tests/e2e/fixtures/cloudwatch_dependencies.py

```diff
@@ -0,0 +1,117 @@
+import subprocess
+import time
+import pytest
+
+from e2e.utils.config import metadata, configure_resource_fixture
+from e2e.conftest import region
+from e2e.fixtures.cluster import cluster
+from e2e.utils.utils import get_eks_client, get_iam_client
+
+
+@pytest.fixture(scope="class")
+def cloudwatch_bootstrap(metadata, region, request, cluster):
+
+    cloudwatch_deps = {"cluster": {"region": region, "name": cluster}}
+
+    def get_eks_nodegroup_role(ClusterName):
+        eks_client = get_eks_client(region)
+        nodegroup = eks_client.list_nodegroups(clusterName=ClusterName)["nodegroups"]
+        nodegroup = eks_client.describe_nodegroup(
+            clusterName=ClusterName, nodegroupName=nodegroup[0]
+        )
+        nodeRole = nodegroup["nodegroup"]["nodeRole"].split("/")[1]
+        return nodeRole
+
+    def attach_cloudwatch_policy(nodeRole):
+        iam_client = get_iam_client(region)
+        iam_client.attach_role_policy(
+            RoleName=nodeRole,
+            PolicyArn="arn:aws:iam::aws:policy/CloudWatchAgentServerPolicy",
+        )
+
+    def detach_cloudwatch_policy(nodeRole):
+        iam_client = get_iam_client(region)
+        iam_client.detach_role_policy(
+            RoleName=nodeRole,
+            PolicyArn="arn:aws:iam::aws:policy/CloudWatchAgentServerPolicy",
+        )
+
+    def install_cloudwatch_container_insights(ClusterName):
+        LogRegion = region
+        FluentBitHttpPort = "2020"
+        FluentBitReadFromHead = "Off"
+        FluentBitHttpServer = "On"
+        FluentBitReadFromTail = "On"
+        cloudwatch_namespace_cmd = []
+        cloudwatch_namespace_cmd += "kubectl apply -f https://raw.githubusercontent.com/aws-samples/amazon-cloudwatch-container-insights/latest/k8s-deployment-manifest-templates/deployment-mode/daemonset/container-insights-monitoring/cloudwatch-namespace.yaml".split()
+        subprocess.call(cloudwatch_namespace_cmd)
+
+        fluent_bit_cmd = []
+        fluent_bit_cmd += f"kubectl create configmap fluent-bit-cluster-info \
+--from-literal=cluster.name={ClusterName} \
+--from-literal=http.server={FluentBitHttpServer} \
+--from-literal=http.port={FluentBitHttpPort} \
+--from-literal=read.head={FluentBitReadFromHead} \
+--from-literal=read.tail={FluentBitReadFromTail} \
+--from-literal=logs.region={LogRegion} -n amazon-cloudwatch".split()
+        subprocess.call(fluent_bit_cmd)
+
+        install_daemon_cmd = []
+        install_daemon_cmd += "kubectl apply -f https://raw.githubusercontent.com/aws-samples/amazon-cloudwatch-container-insights/latest/k8s-deployment-manifest-templates/deployment-mode/daemonset/container-insights-monitoring/fluent-bit/fluent-bit.yaml".split()
+        subprocess.call(install_daemon_cmd)
```

Comment:
Can we get the difference between what you're applying here vs [fluent-bit-quickstart](https://raw.githubusercontent.com/aws-samples/amazon-cloudwatch-container-insights/latest/k8s-deployment-manifest-templates/deployment-mode/daemonset/container-insights-monitoring/quickstart/cwagent-fluent-bit-quickstart.yaml)

recommended in [QuickStart](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Container-Insights-setup-EKS-quickstart.html) page.

Reply:
I would like to see what we are missing by using one vs another.

Reply:
The one in the quickstart page gives us the cloudwatch agent which provides these metrics https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Container-Insights-metrics-EKS.html and for logs it provides the performance of each node of the eks cluster.

Reply:
and what does the existing implementation you have provide ? can you link that as well ?

Reply:
Check the list of log groups in the Region. You should see the following:

    /aws/containerinsights/Cluster_Name/application

    /aws/containerinsights/Cluster_Name/host

    /aws/containerinsights/Cluster_Name/dataplane

is the logs that get uploaded. For metrics it is metrics about the IncomingBytes and IncomingLogEvents etc related to thel ogs being uploaded. For the purpose of testing that Cloudwatch logs/metrics work I think this is sufficient already since we can see that it works and it should as this is default provided by AWS and we are not doing anything special here.

In terms of our documentation I would point towards the full installation which provides the metrics and logs I linked in the previous comment.

Reply:
reading at ur comments it seems there are differences and one is not subset of another right?

Reply:
I am less worried about tests, what you have might be sufficient, I am worried about user documentations.

Reply:
Lets change the current installation to quickstart guide and we should be good to go.